src/test/java/org/broadinstitute/hellbender/engine/FeatureDataSourceUnitTest.java

```diff
@@ -71,6 +72,13 @@ public void testGetSequenceDictionary() {
         }
     }
 
+    @Test (expectedExceptions = GATKException.class)
+    // this test asserts that a helpful exception is thrown for blockZipped files lacking an index as they may not be fully supported
+    //TODO this is a temporary fix until https://github.com/broadinstitute/gatk/issues/4224 has been resolved
```

Comment:
Maybe it is a good idea to add a test that it's failing with unindexed bgzip files, providing wrong results or parsing errors. That test should be disabled and used to test if the issue is solved. Just an idea...

Reply:
Unfortunately implementing that test has problems two-fold, the first is that I can't assert this problem anymore now that I have restricted featureDataSource (well maybe not technically but functionally), and the second is that it's difficult to trigger this for small files. I will put some effort into making some files that trigger the bug, but its inconsistent between machines so I couldn't really enforce much.